sql3/planner/oppqlaggregate.go

```diff
@@ -263,20 +263,33 @@ func (i *pqlAggregateRowIter) Next(ctx context.Context) (types.Row, error) {
 			i.resultValue = int64(actualResult)
 
 		case pilosa.ValCount:
-			if actualResult.DecimalVal == nil {
+			switch i.aggregate.Type().(type) {
+			case *parser.DataTypeInt:
 				if i.aggregate.AggType() == types.AGGREGATE_AVG {
 					average := float64(actualResult.Val) / float64(actualResult.Count)
 					i.resultValue = pql.NewDecimal(int64(average*10000), 4)
 				} else {
 					i.resultValue = int64(actualResult.Val)
 				}
-			} else {
+
+			case *parser.DataTypeDecimal:
 				if i.aggregate.AggType() == types.AGGREGATE_AVG {
-					average := actualResult.DecimalVal.Float64() / float64(actualResult.Count)
-					i.resultValue = pql.NewDecimal(int64(average*10000), 4)
+					if actualResult.DecimalVal == nil {
+						average := float64(actualResult.Val) / float64(actualResult.Count)
+						i.resultValue = pql.NewDecimal(int64(average*10000), 4)
```

Comment:
Should the scale be obtained from the schema?

Reply:
Yeah - good point.